x-pack/plugins/ml/public/application/components/ml_inference/components/test_pipeline.tsx

```diff
@@ -147,27 +226,75 @@ export const TestPipeline: FC<Props> = memo(({ state, sourceIndex }) => {
-                {i18n.translate(
-                  'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.optionalCallout',
-                  { defaultMessage: 'This is an optional step.' }
-                )}
-              </strong>
-              &nbsp;
-              <FormattedMessage
-                id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.description"
-                defaultMessage="Run a simulation of the pipeline to confirm it produces the anticipated results."
-              />{' '}
-              {state.targetField && (
+              {mode === 'step' ? (
+                <>
+                  <strong>
+                    {i18n.translate(
+                      'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.optionalCallout',
+                      { defaultMessage: 'This is an optional step.' }
+                    )}
+                  </strong>
+                  &nbsp;
+                </>
+              ) : null}
+              <>
                 <FormattedMessage
-                  id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.targetFieldHint"
-                  defaultMessage="Check for the target field {targetField} for the prediction in the Result tab."
-                  values={{ targetField: <EuiCode>{state.targetField}</EuiCode> }}
+                  id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.description"
+                  defaultMessage="Run a simulation of the pipeline to confirm it produces the anticipated results. {simulatePipelineDocsLink}"
+                  values={{
+                    simulatePipelineDocsLink: (
+                      <EuiLink external target="_blank" href={links.apis.simulatePipeline}>
+                        Learn more.
+                      </EuiLink>
+                    ),
+                  }}
                 />
+                <br />
+              </>
+              {state.targetField && (
+                <>
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.targetFieldHint"
+                    defaultMessage="Check for the target field {targetField} for the prediction in the Result tab."
+                    values={{
+                      targetField: <EuiCode>{state.targetField}</EuiCode>,
+                    }}
+                  />
+                  <br />
+                </>
               )}
+              {sampleDocsString && sourceIndexMissingError === undefined ? (
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.sourceIndexDocText"
+                  defaultMessage="The provided sample document is taken from the dataset used to train the model."
```

Comment:
Nit - `The provided sample document is taken from the source index used to train the model.`

Reply:
Updated in 2c51898d8c7022455d87e5815e92eb7880e092f9